x-pack/platform/plugins/shared/fleet/public/applications/fleet/hooks/use_agent_policies_with_fips.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useGetAgentsQuery } from '../../../hooks';
+import { AGENTS_PREFIX, SO_SEARCH_LIMIT } from '../constants';
+import type { Agent, AgentPolicy } from '../types';
+import { getNonFipsIntegrations } from '../../../../common/services';
+
+export const hasFipsAgents = (agents: Agent[]) =>
+  agents.some((agent) => agent.local_metadata?.elastic?.agent?.fips === true);
+
+export const useAgentPoliciesWithFipsAgents = (policyIds: string[]) => {
+  const policiesKuery = policyIds
+    .map((policyId) => `${AGENTS_PREFIX}.policy_id:"${policyId}"`)
+    .join(' or ');
+  const { data: fipsAgentsRes } = useGetAgentsQuery(
```

Comment:
With the new Fleet RBAC it's possible someone can have access to agent policies but not agents.

Reply:
It may be interesting to have this server side, maybe an extra property 

```
/agent_policies?withFipsMeta=true
{
  policy1,
  meta: {
    has_fips_agents: true
  }
}
```

Reply:
@nchaulet in https://github.com/elastic/kibana/pull/233134/commits/5ad381d39731b59f62369df7d01b7fd0f7579365 I moved the query that fetches the fips agents to `POST api/fleet/agent_policies/_bulk_get` and followed the implementation of `unprivileged_agents`. 
This way we'll have the total count of fips agents per policy already available everywhere and can be reused if in the future we want to show it in the agent table.